components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -474,6 +491,7 @@ void AdsServiceImpl::StartBatAdsService() {
         bat_ads_service_.BindNewPipeAndPassReceiver(),
         content::ServiceProcessHost::Options()
             .WithDisplayName(IDS_SERVICE_BAT_ADS)
+            .WithExtraCommandLineSwitches(ExtraCommandLineSwitches())
```

Comment:
Have you considered passing the options that you need over to the ads process (through a method on `BatAdsService`, like `SetSysInfo`) instead of passing rewards flags through the command line? Arguably, it might be better to reduce the amount of knowledge the ads process needs to have about the rewards system.

Reply:
We could also parse out the relevant switches (`staging`, `development`) and pass only those. Actually, I was about to ask @tmancey to add these to the `BatLedgerService` utility process, too. WDYT?

Reply:
And we found copying switches to other processes is common in chromium code